packages/yew/tests/use_context.rs

```diff
@@ -254,8 +254,8 @@ async fn use_context_update_works() {
 
     sleep(Duration::ZERO).await;
 
-    // 1 initial render + 3 update steps
-    assert_eq!(obtain_result_by_id("test-0"), "total: 4");
+    // 1 initial render + 1 magic
+    assert_eq!(obtain_result_by_id("test-0"), "total: 2");
```

Comment:
This demonstrates that this pull request is working in reducing unnecessary renders.